iac-common/src/main/java/org/sonar/iac/common/checks/PrivilegeEscalationVector.java

```diff
@@ -21,51 +21,70 @@
+  ATTACH_ROLE_POLICY("Attach Role Policy", List.of("iam:AttachRolePolicy", "sts:AssumeRole")),
+  PUT_USER_POLICY("Put User Policy", List.of("iam:PutUserPolicy")),
+  PUT_GROUP_POLICY("Put Group Policy", List.of("iam:PutGroupPolicy")),
+  PUT_ROLE_POLICY("Put Role Policy", List.of("iam:PutRolePolicy", "sts:AssumeRole")),
+  ADD_USER_TO_GROUP("Add User to Group", List.of("iam:AddUserToGroup")),
+  UPDATE_ASSUME_ROLE_POLICY("Update Assume role Policy", List.of("iam:UpdateAssumeRolePolicy", "sts:AssumeRole")),
+  EC2("EC2", List.of("iam:PassRole", "ec2:RunInstances")),
+  LAMBDA_CREATE_AND_INVOKE("Lambda Create and Invoke", List.of("iam:PassRole", "lambda:CreateFunction", "lambda:InvokeFunction")),
+  LAMBDA_CREATE_AND_ADD_PERMISSION("Lambda Create and Add Permission", List.of("iam:PassRole", "lambda:CreateFunction", "lambda:AddPermission")),
+  LAMBDA_TRIGGERED_WITH_AN_EXTERNAL_EVENT("Lambda triggered with an external event", List.of("iam:PassRole", "lambda:CreateFunction", "lambda:CreateEventSourceMapping")),
+  CLOUD_FORMATION("CloudFormation", List.of("iam:PassRole", "cloudformation:CreateStack")),
+  DATA_PIPELINE("Data Pipeline", List.of("iam:PassRole", "datapipeline:CreatePipeline", "datapipeline:PutPipelineDefinition")),
+  GLUE_DEVELOPMENT_ENDPOINT("Glue Development Endpoint", List.of("iam:PassRole", "glue:CreateDevEndpoint")),
+  UPDATE_GLUE_DEV_ENDPOINT("Update Glue Dev Endpoint", List.of("glue:UpdateDevEndpoint")),
+  UPDATE_LAMBDA_CODE("Update Lambda code", List.of("lambda:UpdateFunctionCode"));
 
   private final List<Permission.SimplePermission> permissions;
+  private final String name;
 
-  PrivilegeEscalationVector(String... permissions) {
-    this.permissions =  Stream.of(permissions).map(Permission.SimplePermission::new).collect(Collectors.toList());
+  PrivilegeEscalationVector(String name, List<String> permissions) {
+    this.name = name;
+    this.permissions = permissions.stream().map(Permission.SimplePermission::new).collect(Collectors.toList());
+  }
+
+  public String getName() {
+    return name;
   }
 
   public boolean isSubsetOf(Collection<Permission> actionPermissions) {
     return permissions.stream().allMatch(p -> actionPermissions.stream().anyMatch(p::isCoveredBy));
   }
 
   public static boolean isSupersetOfAnEscalationVector(Stream<String> actionPermissions) {
+    return getEscalationVector(actionPermissions).isPresent();
+  }
+
+  public static Optional<PrivilegeEscalationVector> getEscalationVector(Stream<String> actionPermissions) {
     Set<Permission> permissionVector = actionPermissions.map(Permission::of).collect(Collectors.toSet());
-    return Stream.of(PrivilegeEscalationVector.values()).anyMatch(vector -> vector.isSubsetOf(permissionVector));
+
+    return Stream.of(PrivilegeEscalationVector.values())
+      .filter(vector -> vector.isSubsetOf(permissionVector))
+      .findFirst();
+  }
+
+  public List<Permission.SimplePermission> permissions() {
```

Comment:
Let's also rename this getter to `getPermissions` to align with the name getter.

Reply:
done